exercise/templates/exercise/_file_link.html

```diff
@@ -1,6 +1,6 @@
 {% load i18n %}
 {% if file.exists %}
-	<a href="{{ file.get_absolute_url }}"{% if not file.is_passed %} class="file-modal"{% endif %} target="_blank">{{ file.filename }}</a>
+	<a href="{{ file.get_absolute_url }}"{% if not file.is_passed %} class="file-modal"{% endif %}>{{ file.filename }}</a>
```

Comment:
Hmm, the `target _blank` attribute probably didn't cause issues in the past because the file modal jQuery plugin overrode the click event handler and target `_blank` was ignored. The language hl click event handler changed the event handlers and the order of them. It probably overrode the file modal handler completely, which is clearly a problem in the language handler.

Reply:
Indeed, The attribute `target=_blank` did not cause any issues in the past but it may be good to make it clear in the code that this link will not open a new tab #716

Reply:
Some browsers could have JavaScript disabled in which case we can not depend on JS to fix the behaviour. In those cases, it would make sense to let the browser open the file in a new tab. Progressive Enhancement also suggests that web pages should remain functional at least for the basic needs even without JS. Accessible technology in particular might disable JS.

In this case, the target attribute has a very small impact. Opening files in new tabs makes some sense with disabled JS but it is not strictly required. What do you think?

https://developer.mozilla.org/en-US/docs/Glossary/Progressive_Enhancement

Reply:
I think we should focus on the browsers we support, such as Firefox, and Chrome. Actually, I am not sure if we mention somewhere the browsers we support. But, if we target A+ to Firefox and Chrome we should have no problem with Javascript. We can also add the following script to the **base.html** template
```html
<noscript>
 For full functionality of this site it is necessary to enable JavaScript.
 Here are the <a href="https://www.enable-javascript.com/">
 instructions how to enable JavaScript in your web browser</a>.
</noscript>
```